### Hipótese 4 (E3)

#### É possível identificar a plataforma de um jogo baseado na sua publisher, no seu número de vendas globais, ano de lançamento, na sua classificação indicativa e no seu gênero?

In [202]:
from IPython.display import display

import pandas as pd
import sklearn as sk
import numpy as np
import scipy as sp
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split, cross_val_score, GridSearchCV
from sklearn.linear_model import LinearRegression
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import explained_variance_score, mean_squared_error
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score
from sklearn import tree

In [203]:
df = pd.read_csv('vgsalesPP2.csv')
display(len(df.columns))
df.head()

645

Unnamed: 0                      Name Platform_left  Year_of_Release  \
0           0                Wii Sports           Wii             2006   
1           1         Super Mario Bros.           NES             1985   
2           2            Mario Kart Wii           Wii             2008   
3           3         Wii Sports Resort           Wii             2009   
4           4  Pokemon Red/Pokemon Blue            GB             1996   

          Genre Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales ...  \
0        Sports  Nintendo  1.000000  1.000000  0.368885     0.799432 ...   
1      Platform  Nintendo  0.703095  0.123619  0.666341     0.072848 ...   
2        Racing  Nintendo  0.379110  0.440608  0.370841     0.311258 ...   
3        Sports  Nintendo  0.377418  0.377417  0.320939     0.279092 ...   
4  Role-Playing  Nintendo  0.272485  0.306975  1.000000     0.094607 ...   

   mixi, Inc  responDESIGN  AO  E E10+  EC  K-A  M  RP  T  
0          0             0   0  1    0   0    0  0   0  0  
1          0             0   0  1    0   0    0  0   0  0  
2          0             0   0  1    0   0    0  0   0  0  
3          0             0   0  1    0   0    0  0   0  0  
4          0             0   0  1    0   0    0  0   0  0  

[5 rows x 645 columns]

In [204]:
lista = []
for i in range(15,46):
    lista.append(df.columns[i])
print(lista)
df.drop(columns=lista,inplace=True)

['2600', '3DO_left', '3DS', 'DC', 'DS', 'GB', 'GBA', 'GC', 'GEN', 'GG', 'N64', 'NES', 'NG', 'PC', 'PCFX', 'PS', 'PS2', 'PS3', 'PS4', 'PSP', 'PSV', 'SAT', 'SCD', 'SNES', 'TG16', 'WS', 'Wii', 'WiiU', 'X360', 'XB', 'XOne']


In [205]:
df_features = df.iloc[:,15:len(df.columns)].copy()
df_features['Global_Sales'] = df['Global_Sales']
df_features['Year'] = df['Year_of_Release']
df_features.head()

Action  Adventure  Fighting  Misc  Platform_right  Puzzle  Racing  \
0       0          0         0     0               0       0       0   
1       0          0         0     0               1       0       0   
2       0          0         0     0               0       0       1   
3       0          0         0     0               0       0       0   
4       0          0         0     0               0       0       0   

   Role-Playing  Shooter  Simulation  ...   AO  E  E10+  EC  K-A  M  RP  T  \
0             0        0           0  ...    0  1     0   0    0  0   0  0   
1             0        0           0  ...    0  1     0   0    0  0   0  0   
2             0        0           0  ...    0  1     0   0    0  0   0  0   
3             0        0           0  ...    0  1     0   0    0  0   0  0   
4             1        0           0  ...    0  1     0   0    0  0   0  0   

   Global_Sales  Year  
0      1.000000  2006  
1      0.487580  1985  
2      0.430389  2008  
3      0.397068  2009  
4      0.380104  1996  

[5 rows x 601 columns]

In [206]:
X = np.array(df_features)
y = np.array(df.Platform_left)

In [207]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [208]:
acertos = []

for i in range(3,31):
    clf_gini = DecisionTreeClassifier(criterion = "gini", random_state = 100,
                               max_depth=i, min_samples_leaf=5)
    clf_gini.fit(X_train, y_train)
    y_pred = clf_gini.predict(X_test)
    taxa = accuracy_score(y_test,y_pred)*100
    print("Taxa de acerto com profundidade {}: {}%".format(i,taxa))
    acertos.append(taxa)

acertos = np.array(acertos)
print("Média de acertos: {}%".format(acertos.mean()))

Taxa de acerto com profundidade 3: 30.795004568991775%
Taxa de acerto com profundidade 4: 34.41973804447152%
Taxa de acerto com profundidade 5: 36.09503502893695%
Taxa de acerto com profundidade 6: 37.465732561681385%
Taxa de acerto com profundidade 7: 39.506548888212%
Taxa de acerto com profundidade 8: 39.71976850441669%
Taxa de acerto com profundidade 9: 40.57264696923546%
Taxa de acerto com profundidade 10: 40.66402680475175%
Taxa de acerto com profundidade 11: 40.63356685957965%
Taxa de acerto com profundidade 12: 40.42034724337496%
Taxa de acerto com profundidade 13: 40.78586658544015%
Taxa de acerto com profundidade 14: 40.11574779165398%
Taxa de acerto com profundidade 15: 39.5979287237283%
Taxa de acerto com profundidade 16: 39.78068839476089%
Taxa de acerto com profundidade 17: 39.71976850441669%
Taxa de acerto com profundidade 18: 39.62838866890039%
Taxa de acerto com profundidade 19: 39.384709107523605%
Taxa de acerto com profundidade 20: 39.384709107523605%
Taxa de acerto c

### Conclusão

A taxa média de acertos foi de 39% aproximadamente para um total de 27 testes cada qual incrementando de 3 à 30 níveis de profundidade na árvore. O resultado foi muito baixo, o que nos leva a refutar a hipótese sobre ser possível prever a plataforma de um jogo baseado na sua Publisher e suas vendas globais.